src/main.rs

```diff
@@ -185,3 +196,53 @@ fn parse_args() -> Result<LndCfg, ArgsError> {
 
     Ok(LndCfg::new(address, cert_file, macaroon_file))
 }
+
+/// Sets the onion messaging feature bit (described in this PR:
+/// https://github.com/lightning/bolts/pull/759/), to signal that we support
+/// onion messaging. This needs to be done every time we start up, because LND
+/// does not currently persist the custom feature bits that are set via the RPC.
+async fn set_feature_bit(mut client: Client) {
+    let onion_feature_bit = 39;
+    let update = tonic_lnd::peersrpc::UpdateFeatureAction {
+        action: i32::from(tonic_lnd::peersrpc::UpdateAction::Add),
+        feature_bit: onion_feature_bit,
+    };
+    let feature_updates = vec![update];
+    let address_updates = vec![];
+
+    let resp = client
+        .peers()
+        .update_node_announcement(tonic_lnd::peersrpc::NodeAnnouncementUpdateRequest {
+            feature_updates,
+            color: String::from(""),
+            alias: String::from(""),
+            address_updates,
+        })
+        .await;
+
+    match resp {
+        Ok(_) => {
+            println!("Now setting the onion messaging feature bit...")
+        }
+        Err(status) => {
+            if status.code() == Code::Unimplemented {
+                panic!("{}", SERVICE_UNIMPLEMENTED);
+            }
+
+            panic!("error updating node announcement {}", status);
+        }
+    };
+
+    // Call get_info again to check the bit was actually set.
+    let info = client
+        .lightning()
+        .get_info(tonic_lnd::lnrpc::GetInfoRequest {})
+        .await
+        .expect("failed to get info");
+
+    if !info.into_inner().features.contains_key(&ONION_FEATURE_BIT) {
+        panic!("onion messaging feature bit failed to be set")
+    }
+
+    println!("Successfully set onion messaging bit");
```

Comment:
Adding `simple_logger::init_with_level(log::Level::Info).unwrap();` in #17, perhaps we can just copy the log setup in here and use `log` rather? That way this PR doesn't need to depend on the other, but we still get nice logging.